### PR TITLE
framework/eventloop: Implement async event handling.

### PR DESCRIPTION
**Pull request by jeongarmy**
Event loop framwork supports an independent loop.
It is started when system start and waits for the event from a task.
If some tasks add event by calling async functions, this event is added a loop of event loop task.
And then event loop task wakes up and handles added event.
 . eventloop_add_timer_async

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [6505a2bdca52c7d97715f69fc045b74fc87a22d2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6505a2bdca52c7d97715f69fc045b74fc87a22d2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4423d9d78c59ec5076226ed2ef38eff0b6178c17] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4423d9d78c59ec5076226ed2ef38eff0b6178c17] - Code Rule Check (C++) OK.

File: framework/include/eventloop/eventloop.h [hunk 1]
```diff
@@ -89,6 +89,27 @@ el_timer_t *eventloop_add_timer(unsigned int timeout, bool repeat, timeout_callb
...
+ * @param[in] timeout interval in milliseconds from the current time of loop for calling callback function
+ * @param[in] repeat the value which represents whether timer runs repeatly or not. \n
+ *            You can stop the periodic timer by calling eventloop_delete_timer
+ * @param[in] func the callback function to be called
+ * @param[in] data data to pass to func when func is called
+ * @return On success, A pointer of created timer is returned if 'repeat' is true. \n
+ *         You should free used timer resources after using timer by calling eventloop_delete_timer with returned pointer. \n
+ *         On the other hand, NULL is returned if 'repeat' is false. \n
+ *         So you don't have to do anything because all resources are freed internally after callback execution. \n
+ *         On failure, NULL is returned
```
Comment (davidfather) [inline framework/include/eventloop/eventloop.h]: How to discriminate two cases: returning NULL on failure and returning NULL on success when repeat is false?

Comment (jeongarmy) [reply framework/include/eventloop/eventloop.h]: Both cases return NULL, so they can distinguished by enabling debug.

**Comment by seinfra** [comment]
Target : [ff8194fc79c3c0f955bd79b0f2a58602e6ee9456] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ff8194fc79c3c0f955bd79b0f2a58602e6ee9456] - Code Rule Check (C++) OK.

File: apps/system/init/init.c [hunk 2]
```diff
@@ -122,6 +125,15 @@ int preapp_start(int argc, char *argv[])
...
+#ifdef CONFIG_EVENTLOOP
+	pid = eventloop_task_start();
+	if (pid <= 0) {
+		printf("eventloop is failed to start, error code is %d\n", pid);
+		goto error_out;
+	}
+#endif
+
+
 #if defined(CONFIG_LIB_USRWORK) || defined(CONFIG_TASH)
```
Comment (sunghan-chang) [inline apps/system/init/init.c]: You should add your config here. line 100 also.

File: framework/include/eventloop/eventloop.h [hunk 3]
```diff
@@ -89,6 +87,25 @@ el_timer_t *eventloop_add_timer(unsigned int timeout, bool repeat, timeout_callb
...
+ * @param[in] repeat the value which represents whether timer runs repeatly or not. \n
+ *            You can stop the periodic timer by calling eventloop_delete_timer
+ * @param[in] func the callback function to be called
+ * @param[in] data data to pass to func when func is called
+ * @return On success, A pointer of created timer is returned. \n
+ *         You should free used timer resources after using timer by calling eventloop_delete_timer with returned pointer. \n
+ *         On failure, NULL is returned
+ * @since TizenRT v2.0 PRE
+ */
+el_timer_t *eventloop_add_timer_async(unsigned int timeout, bool repeat, timeout_callback func, void *data);
```
Comment (sunghan-chang) [inline framework/include/eventloop/eventloop.h]: Could you write difference of free operation against sync command?

**Comment by seinfra** [comment]
Target : [977ff3d648452129417665a054ab638b563428ad] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [13567ae2c6bc8a369a8d9e956628b6a3f3977afc] -

**Comment by seinfra** [comment]
Target : [bc48ad2d75a9296c694ff868daa773e9c69f79c4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bc48ad2d75a9296c694ff868daa773e9c69f79c4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b11f8f433fe54795114e9c43d898a7267a461589] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b11f8f433fe54795114e9c43d898a7267a461589] - Code Rule Check (C++) OK.

**Pull request merged**